src/core/tree.ts

```diff
@@ -46,7 +65,26 @@ export class Tree<T>
     add(where: NodeID, data: T): Tree<T>
     {
         const targetNode = this._selectNodeById(where);
-        targetNode._append(new TreeNode(this.dependencies.idProvider.generate(), data, []));
+
+        let newId = NaN;
+        try
+        {
+            newId = this.dependencies.idProvider.generate();
+        }
+        catch(err)
+        {
+            if (err.name !== "ID_OVERFLOW") 
+            {
+                throw err;
+            }
+
+            this.root._reenumerate(this.dependencies.idProvider.generate.bind(this.dependencies.idProvider));
```

Comment:
I'm seriously considering emulating Python's bound method at the factory level. Does this sound like a good idea?